CardGame.playground/Contents.swift

```diff
@@ -144,5 +202,110 @@ import Foundation
+// Scores are kept - 1 point awarded to winner of each round. (No points awarded if it's a tie.)
+// Game is over when all the cards have been drawn (26 rounds). Winner is announced based on final score.
+
+extension Deck {
+    mutating func drawCard2() -> Card {
+        let cardIndex = Int.random(in: 0..<self.cards.count)
+        let card = self.cards[cardIndex]
+        self.cards.remove(at: cardIndex)
+        return card
+    }
+}
+
+extension CardGameDelegate {
+    func gameDidEnd(player1Score: Int, player2Score: Int) {
+        if player1Score == player2Score {
+            print("The players have tied with a final score of \(player1Score)!")
+        } else if player1Score > player2Score {
+            print("Player 1 has won the game with a final score of \(player1Score).")
+        } else {
+            print("Player 2 has won the game with a final score of \(player2Score).")
+        }
+    }
+}
+
+class CardGameTracker2: CardGameDelegate {
+    func gameDidStart(_ game: CardGame) {
+        if game is HighLow2 {
+            print("Started a new game of High Low - Extended Version.")
+        }
+    }
+    
+    func game(player1DidDraw card1: Card, player2DidDraw card2: Card) {
+        print("Player 1 drew the \(card1) and Player 2 drew the \(card2).")
+    }
+    
+    func gameDidEnd(player1Score: Int, player2Score: Int) {
+        if player1Score == player2Score {
+            print("The players have tied with a final score of \(player1Score).")
+        } else if player1Score > player2Score {
+            print("Player 1 has won the game with a final score of \(player1Score).")
+        } else {
+            print("Player 2 has won the game with a final score of \(player2Score).")
+        }
+    }
+}
+
+class HighLow2: CardGame {
+    var deck = Deck()
+    var delegate: CardGameDelegate?
+    
+    func play() {
+        delegate?.gameDidStart(self)
+        
+        var player1Score = 0
+        var player2Score = 0
+        
+        var player1Card: Card
+        var player2Card: Card
+        
+        for _ in 1...26 {
+            player1Card = deck.drawCard2()
+            player2Card = deck.drawCard2()
+        
+            delegate?.game(player1DidDraw: player1Card, player2DidDraw: player2Card)
+        
+            if player1Card * player2Card {
+                print("Both players drew the \(player1Card.rank). It's a tie! The score is still \(player1Score) to \(player2Score).")
+                print("")
+            } else if player1Card < player2Card {
+                player2Score += 1
+                print("Player 2 has won the round with the \(player2Card). The score is now \(player1Score) to \(player2Score).")
+                print("")
+            } else {
+                player1Score += 1
+                print("Player 1 has won the round with the \(player1Card). The score is now \(player1Score) to \(player2Score).")
+                print("")
+            }
```

Comment:
In case you didn't realize, you can add blank lines with the newline character `\n`

``` swift
if player1Card * player2Card {
    print("Both players drew the \(player1Card.rank). It's a tie! The score is still \(player1Score) to \(player2Score).\n")
} else if player1Card < player2Card {
    player2Score += 1
    print("Player 2 has won the round with the \(player2Card). The score is now \(player1Score) to \(player2Score).\n")
} else {
    player1Score += 1
    print("Player 1 has won the round with the \(player1Card). The score is now \(player1Score) to \(player2Score).\n")
}
```